### PR TITLE
Adds react/avatar

### PR DESCRIPTION
**Pull request by pschoffer**
## Description

This is the first step towards handling avatar change in nicer way. With the ability to crop and rotate.

This step introduces React to the mix. Streamlines the profile picture uploading process.

Inspiration for this is the profile picture change here in github.

## Steps to test

Go to edit my profile and click Change my profile picture:
image

## Next steps (seperate PR)
* Adds the cropping rotation popup

### CODE REVIEW

File: src/Controller/AvatarController.php [hunk 1]
```diff
@@ -55,6 +92,39 @@ public function showAvatar(string $username, string $size): BinaryFileResponse
         return $this->createCacheableResponse($filename);
     }
 
+    private function storeAvatar($memberId, $tmpFilePath)
+    {
+        // TODO
+        // $this->writeMemberphoto($memberId);
```
Comment (pschoffer) [inline src/Controller/AvatarController.php:98]: What is the `membersphotos` Photo DB record for? It seems a bit inconsistent here as we add a record but never remove it when deleting old profile picture. Is that by design?

File: build/members/pages/member.page.php [hunk 2]
```diff
@@ -188,6 +188,14 @@ protected function submenu() {
         $member = $this->member;
         $words = $this->getWords();
         $picture_url = 'members/avatar/'.$member->Username;
+        $globalsJs = json_encode([
+            'baseUrl' => $this->getBaseUrl(),
+            'texts' => [
+                'profile.change.avatar' => $words->get('profile.change.avatar'),
+                'profile.change.avatar.success' => $words->get('profile.change.avatar.success'),
```
Comment (pschoffer) [inline build/members/pages/member.page.php:195]: How do I define the new translations? `profile.change.avatar.success` `profile.change.avatar.fail`

File: build/members/pages/member.page.php [hunk 3]
```diff
@@ -205,8 +213,7 @@ protected function submenu() {
             if ($this->myself) {
                 // TODO : change language code (en) and wordcode
```
Comment (pschoffer) [inline build/members/pages/member.page.php:214]: What does this mean?

File: build/members/pages/member.page.php (hunk 2 above)
Comment (thisismeonmounteverest) [reply build/members/pages/member.page.php:195]: That needs to be done manually to update the database to include the translation IDs.

File: src/Controller/AvatarController.php (hunk 1 above)
Comment (thisismeonmounteverest) [reply src/Controller/AvatarController.php:98]: That is old code. It seems at some point the idea was to upload several profile pictures/images and select one to be used. 

Without an entry in the membersphotos table an empty avatar will be shown.

As you can't decide to not show an avatar/profile picture anymore that entry is never deleted.

File: build/members/pages/member.page.php (hunk 3 above)
Comment (thisismeonmounteverest) [reply build/members/pages/member.page.php:214]: The member page is different if you look at it yourself (myself :)) or at someone else's page.

Comment (pschoffer) [reply build/members/pages/member.page.php:214]: Yeah I got that part, but the `TODO: change language code (en) and wordcode` ?

File: build/members/pages/member.page.php (hunk 2 above)
Comment (pschoffer) [reply build/members/pages/member.page.php:195]: Do I do that or somebody does that after merge? :)

File: src/Controller/AvatarController.php (hunk 1 above)
Comment (pschoffer) [reply src/Controller/AvatarController.php:98]: Oh but looking at the AvatarController:showAvatar that is not used nowadays... so it is probably safe to delete, right?

**Comment by crumbking** [comment]
> Do I do that or somebody does that after merge? :) 

Can be done later on beta. Locally you could use user bwadmin. He has translation rights. In the translation tool in the submenu is the create translation link.

File: src/Controller/AvatarController.php (hunk 1 above)
Comment (thisismeonmounteverest) [reply src/Controller/AvatarController.php:98]: We shoudl check all usages in queries for membersphotos and then delete it after review.

File: build/members/pages/member.page.php (hunk 2 above)
Comment (thisismeonmounteverest) [reply build/members/pages/member.page.php:195]: That's normally done through the translation interface. So I can take that.

File: build/members/pages/member.page.php (hunk 3 above)
Comment (thisismeonmounteverest) [reply build/members/pages/member.page.php:214]: Can't say that comment is really, really old.

**Pull request merged**